pykokkos/lib/ufuncs.py

```diff
@@ -959,17 +960,166 @@ def multiply(viewA, viewB):
+    Check whether two views are broadcastable as defined here:
+    https://numpy.org/doc/stable/user/basics.broadcasting.html
+
+    Parameters
+    ----------
+    viewA : pykokkos view
+            Input view.
+    viewB : pykokkos view
+            Input view.
+    
+    Returns
+    -------
+    _ : boolean
+           True if both views are compatible.
+    """
+
+    if viewA.shape == viewB.shape:
+        return False # cannot broadcast same dims
+
+    v1_p = len(viewA.shape) -1
+    v2_p = len(viewB.shape) -1
+
+    while v1_p > -1 and v2_p > -1:
+        if viewA.shape[v1_p] != viewB.shape[v2_p]:
+            if viewA.shape[v1_p] != 1 and viewB.shape[v2_p] != 1:
+                return False
+        
+        v1_p -= 1
+        v2_p -= 1
+    
+    return True
+
+@pk.workunit
+def stretch_fill_impl_scalar_into_1d(tid, scalar, viewOut):
+        viewOut[tid] = scalar
+
+@pk.workunit
+def stretch_fill_impl_scalar_into_2d(team_member, cols, scalar, viewOut):
+    tid: int = team_member.league_rank()
+    def row_fill(i: int):
+        viewOut[tid][i] = scalar
+    pk.parallel_for(pk.TeamThreadRange(team_member, cols), row_fill)
+    
+@pk.workunit
+def stretch_fill_impl_1d_into_2d(team_member, cols, viewIn, viewOut):
+    tid: int = team_member.league_rank()
+    def row_fill(i: int):
+        viewOut[tid][i] = viewIn[i]
+    pk.parallel_for(pk.TeamThreadRange(team_member, cols), row_fill)
+
+@pk.workunit
+def stretch_fill_impl_2d(team_member, inner_its, col_wise, viewIn, viewOut):
+    tid: int = team_member.league_rank()
+    def inner_fill(i: int):
+        if col_wise:
+            viewOut[i][tid] = viewIn[i][0]
+        else:
+            viewOut[tid][i] = viewIn[0][i]
+    pk.parallel_for(pk.TeamThreadRange(team_member, inner_its), inner_fill)
+
+        
+
+def broadcast_view(val, viewB):
+    """
+    Broadcasts val onto viewB, returns the "stretched" version of viewA
+
+    Parameters
+    ----------
+    val : pykokkos view or Scalar
+            View or scalar to be broadcasted (is shorter and compatible in dimensions).
+    viewB : pykokkos view
+            View to be broadcasted onto (is longer and compatible in dimensions).
+
+    Returns
+    -------
+    out : pykokkos view
+           Broadcasted version of viewA.
+
+    """
+    if len(viewB.shape) > 2:
+        raise NotImplementedError("Broadcasting is only supported upto 2D views")
+
+    is_view = False
+    if isinstance(val, ViewType):
+        for dim in val.shape:
+            if dim != 1:
+                is_view = True
+        
+        if not is_view:
+            val = val[0] if len(val.shape) == 1 else val[0][0]
+
+    if is_view:
+        if not check_broadcastable_impl(val, viewB) or not val.shape < viewB.shape:
+            raise ValueError("Incompatible broadcast")
+        if not val.dtype == viewB.dtype: 
+            raise ValueError("Broadcastable views must have same dtypes")
+
+    out = pk.View([dim for dim in viewB.shape], viewB.dtype)
```

Comment:
you should just be able to do `pk.View(viewB.shape, viewB.dtype)`

Reply:
Let me give this a try, I thought view.shape returns a tuple and the constructor takes a list